components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtil.java

```diff
@@ -2000,4 +2017,120 @@ public static SAMLSSOServiceProviderDO getSPConfig(String tenantDomain, String i
             PrivilegedCarbonContext.endTenantFlow();
         }
     }
+
+    public static LogoutRequest buildLogoutRequest(SAMLSSOServiceProviderDO serviceProviderDO, String subject,
+                                                   String sessionId) throws IdentityException {
+
+        SingleLogoutRequestDTO logoutReqDTO = new SingleLogoutRequestDTO();
+
+        if (StringUtils.isNotBlank(serviceProviderDO.getSloRequestURL())) {
+            logoutReqDTO.setAssertionConsumerURL(serviceProviderDO.getSloRequestURL());
+        } else if (StringUtils.isNotBlank(serviceProviderDO.getSloResponseURL())) {
+            logoutReqDTO.setAssertionConsumerURL(serviceProviderDO.getSloResponseURL());
+        } else {
+            logoutReqDTO.setAssertionConsumerURL(serviceProviderDO.getAssertionConsumerUrl());
+        }
+
+        LogoutRequest logoutReq = new LogoutRequestBuilder().buildObject();
+
+        logoutReq.setID(SAMLSSOUtil.createID());
+
+        String destination = logoutReqDTO.getAssertionConsumerURL();
+        if (destination != null) {
+            logoutReq.setDestination(destination);
+        }
+
+        DateTime issueInstant = new DateTime();
+        logoutReq.setIssueInstant(issueInstant);
+        logoutReq.setIssuer(SAMLSSOUtil.getIssuerFromTenantDomain(serviceProviderDO.getTenantDomain()));
+        logoutReq.setNotOnOrAfter(new DateTime(issueInstant.getMillis() + 5 * 60 * 1000));
```

Comment:
Let's not have magic numbers. Always create constants for these kinds of things. (Five minutes right ?)